File: frontend/src/api/guidesBridge.ts
import { apiFetch } from "./client";

export interface GuidesBridgeConfig {
  enabled: boolean;
  company_slug: string;
  bridge_email: string;
  password_configured: boolean;
  configured_in_database: boolean;
  api_url: string;
  web_url: string;
  proxy_web_url: string;
}

export interface GuidesBridgeConfigUpdate {
  company_slug: string;
  bridge_email: string;
  password?: string;
}

export interface GuidesBridgeTestResult {
  success: boolean;
  message: string;
  email?: string | null;
  role?: string | null;
}

export function fetchGuidesBridgeConfig(token: string): Promise<GuidesBridgeConfig> {
  return apiFetch<GuidesBridgeConfig>("/v1/guides/bridge-config", token);
}

export function updateGuidesBridgeConfig(
  token: string,
  payload: GuidesBridgeConfigUpdate,
): Promise<GuidesBridgeConfig> {
  return apiFetch<GuidesBridgeConfig>("/v1/guides/bridge-config", token, {
    method: "PUT",
    body: JSON.stringify(payload),
  });
}

export function testGuidesBridgeConfig(token: string): Promise<GuidesBridgeTestResult> {
  return apiFetch<GuidesBridgeTestResult>("/v1/guides/bridge-config/test", token, {
    method: "POST",
  });
}

File: services/identity/app/schemas/guides_bridge.py
from pydantic import BaseModel, Field


class GuidesBridgeConfigResponse(BaseModel):
    enabled: bool
    company_slug: str
    bridge_email: str
    password_configured: bool
    configured_in_database: bool
    api_url: str
    web_url: str
    proxy_web_url: str


class GuidesBridgeConfigUpdate(BaseModel):
    company_slug: str = Field(min_length=2, max_length=64)
    bridge_email: str = Field(min_length=3, max_length=255)
    password: str | None = Field(default=None, min_length=1, max_length=256)


class GuidesBridgeTestResponse(BaseModel):
    success: bool
    message: str
    email: str | None = None
    role: str | None = None
